future/transaction.go

```diff
@@ -139,6 +139,73 @@ func MakeKeyRegTxn(account string, note []byte, params types.SuggestedParams, vo
 	return setFee(tx, params)
 }
 
+// MakeKeyRegTxnV2 constructs a keyreg transaction using the passed parameters.
+// - account is a checksummed, human-readable address for which we register the given participation key.
+// - note is a byte array
+// - params is typically received from algod, it defines common-to-all-txns arguments like fee and validity period
+// KeyReg parameters:
+// - votePK is a base64-encoded string corresponding to the root participation public key
+// - selectionKey is a base64-encoded string corresponding to the vrf public key
+// - voteFirst is the first round this participation key is valid
+// - voteLast is the last round this participation key is valid
+// - voteKeyDilution is the dilution for the 2-level participation key
+// - nonpart is an indicator marking a key registration participating or nonparticipating
+// - stateProofPK is a base64-encoded string corresponding to the block proof public key
+func MakeKeyRegTxnV2(account string, note []byte, params types.SuggestedParams, voteKey, selectionKey string, voteFirst, voteLast, voteKeyDilution uint64, nonpart bool, stateProofPK types.Verifier) (types.Transaction, error) {
```

Comment:
I think `MakeKeyRegTxnWithStateProofKey` would be a better name.

Also, similar to `voteKey` and `selectionKey`, can `stateProofPK` be a string instead, then you convert it to the right type in the function? Also you might want to put it right after `selectionKey`, since this is an entirely new function and the order of arguments is more flexible.